### ISSUE TITLE
Row stay in the selected data after they are deleted 

### ISSUE THREAD

**Issue by Paraplegix** [olifolkerd/tabulator#229]
jsfiddle for the demo
https://jsfiddle.net/da60e48j/4/

1. Add rows using the buttons, 
2. select some rows : the selected row counter should go up (number at the right of the buttons),
3. clic the "delete rows" button to delete only selected row
4. The selected row counter doesn't change, and selecting more row will only increase the counter

Even by using $().tabulator("deselectRow") will not deselect previously deleted row
The deleted selected row seem to be stuck in the selected data forever :cry: 
You can use the "show selected id" to see what the .tabulator("getSelectedData") returns.

The jsfiddle is configured with selectable:true, but i've done test with other settings :

- Tested with `selectable:4 `=> another issue here : tabulator completly ignore max selectable limitations if you delete one or more previously selected row
- Tested with `selectable:4, selectableRollingSelection:false` : counter can't go over the max selectable , but you if you delete 2 selected rows, you can only select 2 other rows, if you delete 4 selected row you can't select any row anymore etc.
- Tested with `selectablePersistence:false` : Appart from the fantom rows in the selectedData, no other odd behavior (compared to selectable:X and/or Rolling selection)

This is not a major issue, but can be annoying as selected row can be used for row deletion sometimes :disappointed: 

A simple workaround before fix is simply deselecting the row before deletion by putting this line before every "deleteRow" action :
`$("#table").tabulator("deselectRow", rowId);`
`$("#table").tabulator("deleteRow", rowId)`

I don't know if i didn't put a bit too much info for this issue tho :confounded: 

Ho by the way, i've noticed a typo in the documentation : in the Selectable Rows section, for the code example of Rolling Selection and Persistent Selection both have the same comment : "`//disable rolling selection`" where for Persistent Selection it should be something like "`//disable persistent selection`"

**Comment by olifolkerd** [comment]
Hey,

Good spot! i hadnt considered that possibility, i will get in a fix this weekend.

Cheers

Oli

**Comment by olifolkerd** [comment]
Hey,

This fix has now been pushed to the repo.

I did have to make a slight change to your deselectrow button click binding, because it uses the selected array which is passed by reference when one row was deleted it was removed from the array causing the foreach to skip every other row. The code below clones the array to prevent this from happening.

```js
$("#deleterow").click(function(){
	var selected = $.extend(true, [], $("#example-table").tabulator("getSelectedData"))
	
	selected.forEach(function(row){
		$("#example-table").tabulator("deleteRow", row.id);
	})
});
```

Cheers

Oli
